packages/xstate-wallet/src/utils/outcome.ts

```diff
@@ -1,51 +1,14 @@
-import {
-  AllocationItem,
-  SimpleEthAllocation,
-  Outcome,
-  SimpleTokenAllocation,
-  SimpleTokenGuarantee,
-  MixedAllocation,
-  SimpleEthGuarantee
-} from '../store/types';
-
-const outcomeGuard = <T extends Outcome>(type: Outcome['type']) => (o: Outcome): o is T =>
-  o.type === type;
-export const isSimpleEthAllocation = outcomeGuard<SimpleEthAllocation>('SimpleEthAllocation');
-export const isSimpleEthGuarantee = outcomeGuard<SimpleEthGuarantee>('SimpleEthGuarantee');
-export const isSimpleTokenAllocation = outcomeGuard<SimpleTokenAllocation>('SimpleTokenAllocation');
-export const isSimpleTokenGuarantee = outcomeGuard<SimpleTokenGuarantee>('SimpleTokenGuarantee');
-export const isMixedAllocation = outcomeGuard<MixedAllocation>('MixedAllocation');
-
-export const simpleEthAllocation = (...allocationItems: AllocationItem[]): SimpleEthAllocation => ({
-  type: 'SimpleEthAllocation',
+import {AllocationItem, SimpleAllocation, Outcome} from '../store/types';
+import {ETH_ASSET_HOLDER_ADDRESS} from '../constants';
+
+export function isSimpleEthAllocation(outcome: Outcome): outcome is SimpleAllocation {
+  return (
+    outcome.type === 'SimpleAllocation' && outcome.assetHolderAddress === ETH_ASSET_HOLDER_ADDRESS
```

Comment:
What was the benefit of removing the `SimpleEthAllocation` type and replacing it with this check? It seems like we still have a union type (eth allocations are fundamentally different from token allocations, since depositing is different) except that the compiler doesn't know about the union type (this type guard, `isSimpleEthAllocation`, does not assert that the input is a `SimpleEthAllocation`, but instead is a `SimpleAllocation`)

Reply:
This would make a bit more sense if this type guard checked the _token address_, but the fact that `constants.ts` exports a `ETH_ASSET_HOLDER_ADDRESS` muddles things a bit -- the intention of this PR is that there might not be a unique such address.

Reply:
The intention really was to use `assetHolderAddress` instead of `token` in the wire format, and therefore also internally in the wallet. Once this change happens, there's no difference internally* between `SimpleEthAllocation` and `SimpleTokenAllocation`, so I decided to unify these types into `SimpleAllocation`. 

While making this change, I noticed that the protocols were using `isSimpleEthAllocation` to actually restrict the supported token type to eth. I didn't want to change that behaviour as part of the PR, so I ported the method across to work with the new formats. The intention is that `isSimpleEthAllocation` won't exist in the long-term, once we allow the wallet to support other tokens.

\* As you point out, there _are_ differences when it comes to interacting with the blockchain, My assumption was that these differences only manifest in the deposit transaction sent by the `Chain`, so it can do a last-minute switch on the asset holder address to accomplish this.